src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -1351,50 +1378,74 @@ public boolean accept(File file) {
+                onAlteration(executor, parentPath, file, true, ChangeType.CREATE);
             }
 
             @Override
             public void onDirectoryDelete(File file) {
-                onAlteration(executor, watcher, parentPath, file, true, ChangeType.DELETE);
+                onAlteration(executor, parentPath, file, true, ChangeType.DELETE);
             }
 
             @Override
             public void onDirectoryChange(File file) {
-                onAlteration(executor, watcher, parentPath, file, true, ChangeType.MODIFY);
+                onAlteration(executor, parentPath, file, true, ChangeType.MODIFY);
             }
-            
+
             @Override
             public void onFileCreate(File file) {
-                onAlteration(executor, watcher, parentPath, file, false, ChangeType.CREATE);
+                onAlteration(executor, parentPath, file, false, ChangeType.CREATE);
             }
 
             @Override
             public void onFileDelete(File file) {
-                onAlteration(executor, watcher, parentPath, file, false, ChangeType.DELETE);
+                onAlteration(executor, parentPath, file, false, ChangeType.DELETE);
             }
 
             @Override
             public void onFileChange(File file) {
-                onAlteration(executor, watcher, parentPath, file, false, ChangeType.MODIFY);
+                onAlteration(executor, parentPath, file, false, ChangeType.MODIFY);
             }
 
-            private void onAlteration(final ThreadPoolExecutor executor, final WatchService watcher,
-                    final String parentPath, File file, boolean isDirectory, ChangeType changeType) {
+            private void onAlteration(final ThreadPoolExecutor executor, final String parentPath, File file,
+                    boolean isDirectory, ChangeType changeType) {
+                if (trackingMode == FileTrackMode.NOT_SET) {
+                    try {
+                        WatchKey wk = null;
+                        if (watcher != null) {
+                            wk = watcher.poll(100, TimeUnit.MILLISECONDS);
+                        }
+                        List<WatchEvent<?>> events = null;
+                        if (wk != null) {
+                            events = wk.pollEvents();
+                        }
+                        if ((events == null) || events.isEmpty()) {
+                            trackingMode = FileTrackMode.POLLING;
+                            if (watcher != null) {
+                                watcher.close();
+                            }
+                        } else {
+                            trackingMode = FileTrackMode.FILE_WATCHER;
+                            disablePolling();
+                        }
+                    } catch (Exception e) {
+                        // TODO Auto-generated catch block
+                        e.printStackTrace();
```

Comment:
Log the exception using `error(.....)`

Reply:
How is this?
```
catch (Exception e) {
                        error("An error occured attempting to retrieve the watch key or close the file watcher. " + e.getMessage());
                    }
```

Reply:
Also include the throwable as the second parameter so the stacktrace can be logged. e.g. use `error(String msg, Throwable e)`